internal/cephfs/store/volumeoptions.go

```diff
@@ -323,6 +327,32 @@ func NewVolumeOptions(
 	return &opts, nil
 }
 
+func IsVolumeCreateRO(caps []*csi.VolumeCapability) bool {
+	for _, cap := range caps {
+		if cap.AccessMode != nil {
+			switch cap.AccessMode.Mode { //nolint:exhaustive // only check what we want
+			case csi.VolumeCapability_AccessMode_MULTI_NODE_READER_ONLY,
+				csi.VolumeCapability_AccessMode_SINGLE_NODE_READER_ONLY:
+				return true
+			}
+		}
+	}
+
+	return false
+}
+
+func IsDataSourceSnapshot(volumeSource *csi.VolumeContentSource) bool {
```

Comment:
VolumeSource can itself be nil too,
for a bare createPVC, this can lead to a panic here